src/components/dropdown-auto-complete.js

```diff
@@ -0,0 +1,277 @@
+    startIcon: 'mr-2.5 fill-current stroke-current',
+    endIcon: 'ml-2.5 fill-current stroke-current',
+    selected: 'font-semibold text-secondary-900 bg-interactive-100 hover:text-secondary-900 hover:bg-interactive-100',
+  })
+
+  const dropdownClasses = Object.freeze({
+    root: classes.root,
+    menu: `${!data.length > 0 && 'hidden'} ${classes.menu ? classes.menu : 'w-250px'}`,
+    content: classes.content,
+  })
+
+  const textFieldClasses =  Object.freeze({
+    container: classes.inputContainer,
+    input: `capitalize ${classes.input}`,
+  })
+
+  useEffect(() => {
+    const initialOptions = []
+
+    data && data.forEach((el) => {
+      el.items.forEach((item) => {
+        initialOptions.push(item)
+      })
+    })
+    setSelectedOptions([])
+    setOptions(initialOptions)
+  }, [data])
+
+  if (!componentIsActive && open) {
+    setOpen(!open)
+  }
+
+  const onClickSelect = () => {
+    setComponentIsActive((state) => !state)
+    setOpen(!open)
+  }
+
+  const renderList = (data) => (
+    <>
+      {data.map((item, index) => {
+        return (
+          <div 
+            key={`item-container-${index}`} 
+            className={`item-container-${index} 
+              ${dropdownSelectClasses.itemContainer}
+              ${filteredOptions.length && !filteredOptions.includes(item) && 'hidden'} 
+            `} 
+            onClick={() => handleOnClick(index, item)}
+          >
+            <div 
+              className={`content-container-${index}
+                ${dropdownSelectClasses.contentContainer}
+                ${selectedOptions === item && dropdownSelectClasses.selected} 
+              `}
+            >
+              {renderListItem(item)}
+              {item.description && <div className={`description-container-${index} ${dropdownSelectClasses.description}`}>{item.description}</div>}
+            </div>
+          </div>
+        )
+      })}
+    </>
+  )
+
+  const renderListItem = (item) => {
+    let selected = null
+
+    return (
+      <div className={dropdownSelectClasses.contentHeader}>
+        <div className='flex flex-row items-center'>
+          {item.startIcon && <div className={dropdownSelectClasses.startIcon}>{item.startIcon}</div>}
+          <span>{item.title || item.type}</span>
+          {item.endIcon && <div className={dropdownSelectClasses.endIcon}>{item.endIcon}</div>}
+        </div>
+        {selected}
+      </div>
+    )
+  }
+
+  const handleOnClick = (i, value) => {
+    if (selectedOptions === value) {
+      setSelectedOptions([])
+    } else {
+      setSelectedOptions(value)
+      onClickSelect()
+    }
+    
+    onSelect({ ...value, i })
+    setUserInput(value.title)
+    setFilteredOptions([])
+  }
+
+  const onChange = (val) => {
+    const filteredSuggestions = options.filter(
+      suggestion =>
+        suggestion.title.toLowerCase().indexOf(val.toLowerCase()) > -1,
+    )
+    setFilteredOptions(filteredSuggestions)
+    !val && setSelectedOptions([])
```

Comment:
if the evaluation result is not assigned in a variable, prefer to use explicit flow control:

```js
if (!val) {
  setSelectedOptions([])
}
```